### PR TITLE
API-41957-poa-as-self-check

### PR DESCRIPTION
**Pull request by rockwellwindsor-va**
## Summary
* Skips dependent workflow is the `relationship` in the `Claimant` object is set to `Self`
* Adds/Updates tests

## Related issue(s)
[API-41957](https://jira.devops.va.gov/browse/API-41957)

## Testing done

- [x] *New code is covered by unit tests*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
	modified:   modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb
	modified:   modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb
	modified:   modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb
	modified:   modules/claims_api/spec/requests/v1/forms/2122_spec.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/power_of_attorney/2122_spec.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/power_of_attorney/2122a_spec.rb

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb [hunk 1]
```diff
@@ -170,13 +170,22 @@ def update_auth_headers_for_dependent(poa, claimant_information)
...
+
+            relationship = form_attributes['claimant']['relationship']&.downcase
+            # even with both conditions being met
+            # this is the veteran if sent w/ relationship of 'Self'
+            # so we need to verify this is also true
+            relationship != 'self'
+          else
+            false
+          end
         end
```
Comment (tycol7) [inline modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb:185]:
```suggestion
        def allow_dependent_claimant?
          return false unless Flipper.enabled?(:lighthouse_claims_api_poa_dependent_claimants)

          claimant = form_attributes['claimant']
          
          claimant.present? && claimant['relationship']&.downcase != 'self'
        end
```

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb [hunk 2]
```diff
@@ -64,8 +64,18 @@ def shared_form_validation(form_number)
...
+             form_attributes['claimant'].present?
+
+            relationship = form_attributes['claimant']['relationship']&.downcase
+            # even with both conditions being met
+            # this is the veteran if sent w/ relationship of 'Self'
+            # so we need to verify this is also true
+            relationship != 'self'
+          else
+            false
+          end
```
Comment (tycol7) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb:78]: Same suggestion as above.

File: modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb [hunk 3]
```diff
@@ -18,12 +18,22 @@ def validate_form_2122_and_2122a_submission_values(user_profile:, veteran_partic
...
+           form_attributes['claimant'].present?
+
+          relationship = form_attributes['claimant']['relationship']&.downcase
+          # even with both conditions being met
+          # this is the veteran if sent w/ relationship of 'Self'
+          # so we need to verify this is also true
+          relationship != 'self'
+        else
+          false
+        end
```
Comment (tycol7) [inline modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb:32]: Same suggestion as above. Is it worth centralizing this logic somewhere (to keep the code DRY)?

**Review by tycol7 — COMMENTED** [review]
looking good, one code suggestion and a question about consolidating the logic.

File: modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb (hunk 3 above)
Comment (rockwellwindsor-va) [reply modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb:32]: Ya I agree, I was struggling to think of a good place for it between the two versions, but I will think on it and see what I can get

File: modules/claims_api/spec/requests/v2/veterans/power_of_attorney/2122a_spec.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@
 require 'token_validation/v2/client'
 require 'bgs_service/local_bgs'
 
-RSpec.describe 'ClaimsApi::V1::PowerOfAttorney::2122a', type: :request do
+RSpec.describe 'ClaimsApi::V2::PowerOfAttorney::2122a', type: :request do
```
Comment (rockwellwindsor-va) [inline modules/claims_api/spec/requests/v2/veterans/power_of_attorney/2122a_spec.rb:8]: This is an unrelated clean up.  This was a typo left over from the big test suite refactor.

File: modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb (hunk 3 above)
Comment (rockwellwindsor-va) [reply modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb]: broken that out into its own file, not sure how much, if at all, that may expand so it's own module _may_ be a bit much let me know what you think about the approach to unifying this method across version.

**Review by FonzMP — APPROVED** [review]
:shipit:

**Review by rmtolmach — APPROVED** [review]
